File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.js
// DFS와 BFS

const input = require("fs")
  .readFileSync("/dev/stdin")
  .toString()
  .trim()
  .split("\n");

const [N, M, V] = input[0].split(" ").map(Number);

//인접행렬 구하기
const graph = Array.from(Array(N + 1), () => new Array(N + 1).fill(0));

for (let i = 1; i <= M; i++) {
  let [row, column] = input[i].split(" ").map(Number);
  graph[row][column] = 1;
  graph[column][row] = 1;
}

const DFS_visited = new Array(N + 1).fill(false);
const DFS_answer = [];
const BFS_visited = new Array(N + 1).fill(false);
const BFS_answer = [];

function DFS(V) {
  DFS_visited[V] = true;
  DFS_answer.push(V);
  for (let i = 1; i < graph.length; i++) {
    if (graph[V][i] === 1 && DFS_visited[i] === false) {
      DFS(i);
    }
  }
}

function BFS(V) {
  const queue = [];
  queue.push(V);
  BFS_answer.push(V);

  while (queue.length !== 0) {
    let dequeue = queue.shift();
    BFS_visited[V] = true;
    for (let i = 1; i < graph.length; i++) {
      if (graph[dequeue][i] === 1 && BFS_visited[i] === false) {
        BFS_visited[i] = true;
        queue.push(i);
        BFS_answer.push(i);
      }
    }
  }
}

DFS(V);
BFS(V);

console.log(DFS_answer.join(" "));
console.log(BFS_answer.join(" "));

File: 백준/Silver/1260. DFS와 BFS/DFS와 BFS.py

N, M, V = map(int,input().split())

adj_list = list([] for _ in range(N+1))

for el in range(M):
    start, end = map(int, input().split())
    adj_list[start].append(end)
    adj_list[end].append(start)

# adj 오름차순 정렬
for el in adj_list:
    el.sort()



dfs_answer=[]
dfs_visited=[0]*(N+1)

def dfs(vertex):
    # 방문처리
    dfs_visited[vertex] = 1
    dfs_answer.append(vertex)

    for next in adj_list[vertex]:
        if not dfs_visited[next]:
            dfs(next)

dfs(V)


bfs_answer=[]
bfs_visited=[0]*(N+1)

def bfs(vertex):
    # 초기값 처리
    queue = []
    queue.append(vertex)
    bfs_visited[vertex] = 1
    bfs_answer.append(vertex)

    while queue:
        v = queue.pop(0)

        for el in adj_list[v]:
            if not bfs_visited[el]:
                bfs_visited[el] = 1
                bfs_answer.append(el)        
                queue.append(el)


bfs(V)


print(*dfs_answer)
print(*bfs_answer)